src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/ExternalMovementsSyncService.kt

```diff
@@ -583,26 +583,36 @@ class ExternalMovementsSyncService(
     .let { nomisMovement ->
       val dpsOccurrenceId = nomisMovement.scheduledTemporaryAbsenceId?.let { requireParentScheduleExists(it) }
         ?.also { telemetry["dpsOccurrenceId"] = it }
-      val dpsMovementId = dpsApiService.syncTapMovement(prisonerNumber, nomisMovement.toDpsRequest(id = existingDpsMovementId, occurrenceId = dpsOccurrenceId)).id
+
+      val (address, uprn) = deriveDpsAddress(prisonerNumber, existingMapping, nomisMovement.toAddressId, nomisMovement.toAddressOwnerClass, nomisMovement.toFullAddress)
+      uprn?.also { telemetry["dpsUprn"] = it }
+      val dpsMovement = nomisMovement.toDpsRequest(existingMapping?.dpsMovementId, dpsOccurrenceId, address, uprn)
```

Comment:
This is the main change (and similar change for inbound movements) - we attempt to find the DPS UPRN address ID in `deriveDpsAddress` if we've seen it before.